File: Genius.Client/ClientApp/src/genius/SolverResponse.ts
/**
 * This Source Code Form is subject to the terms of the GNU GPL-3.0 License.
 * If a copy of the GPL-3.0 was not distributed with this file, You can obtain one at https://www.gnu.org/licenses/gpl-3.0.en.html.
 * Copyright (C) 2022 Leszek Pomianowski.
 * All Rights Reserved.
 */
import ISolverResponse from './interfaces/ISolverResponse';
import IExpertCondition from './interfaces/IExpertCondition';
import IExpertProduct from './interfaces/IExpertProduct';

export default class SolverResponse implements ISolverResponse {
  public systemId: number = 0;

  public isSolved: boolean = false;

  public status: number = 0;

  public nextCondition?: IExpertCondition;

  public products: IExpertProduct[] = [];

  public constructor(
    systemId: number,
    isSolved: boolean,
    status: number,
    products: IExpertProduct[],
    nextCondition: IExpertCondition | null = null,
  ) {
    this.systemId = systemId;
    this.isSolved = isSolved;
    this.status = status;
    this.products = products;

    if (nextCondition !== null) {
      this.nextCondition = nextCondition;
    }
  }
}

File: Genius/Expert/SolverResponse.cs
// This Source Code Form is subject to the terms of the GNU GPL-3.0 License.
// If a copy of the GPL-3.0 was not distributed with this file, You can obtain one at https://www.gnu.org/licenses/gpl-3.0.en.html.
// Copyright (C) 2022 Leszek Pomianowski.
// All Rights Reserved.

using Genius.Data.Models.Expert;
using Genius.Expert.Interfaces;
using System.Collections.Generic;

namespace Genius.Expert
{
    public class SolverResponse : ISolverResponse
    {
        public int SystemId { get; set; } = 0;

        public bool IsMultiple { get; set; } = false;

        public bool IsSolved { get; set; } = false;

        public SolverStatus Status { get; set; } = SolverStatus.Unknown;

        public IEnumerable<Condition> NextConditions { get; set; } = new List<Condition>();

        public IEnumerable<Product> ResultingProducts { get; set; } = new List<Product>();
    }
}
